### PR TITLE
[3007.x] Add Linode Networking Interfaces Support in Salt-Cloud

### LINKED ISSUE #65908: [FEATURE REQUEST] Linode Instance Networking Interfaces Support During VM Creation in Salt-Cloud
When creating a Linode instance as a VM via `salt-cloud`, it would be much nicer to support specifying network interfaces, like a VPC interface on `eth0`. This can be done by parsing `interfaces` list in a Linode profile and passing it to the Linode APIv4 during creation operation.

Sample profile file:

```yaml
my-linode-profile:
  provider: my-linode-provider
  size: g6-nanode-1
  image: linode/ubuntu22.04
  location: us-mia
  interfaces:
    - purpose: public
    - purpose: vlan
      label: cool-vlan
    - purpose: vpc
      subnet_id: 1234
      ipv4:
        vpc: 10.0.4.10
```

### PR DESCRIPTION
**Pull request by zliang-akamai**
### What does this PR do?
This PR is to allow user to specify networking interfaces configuration of the Linode instance (Cloud VM) during creation.

### What issues does this PR fix or reference?
Fixes: #65908 

### Previous Behavior
Explicit definition of Linode networking interfaces are not supported during a Linode VM creation.

### New Behavior
Explicit definition of Linode networking interfaces are now supported during a Linode VM creation.

### Merge requirements satisfied?

- [x] Docs
- [x] Changelog - https://docs.saltproject.io/en/master/topics/development/changelog.html
- [x] Tests written/updated

### Commits signed with GPG?
Yes

### CODE REVIEW

File: doc/topics/cloud/linode.rst [hunk 1]
```diff
@@ -105,7 +105,31 @@ Configuration Options
...
+        VLAN interface.
+
+        ``subnet_id`` and ``ipv4`` are only allowed for a VPC interface, and ``subnet_id`` is required for a
+        VPC interface.
+
+        ``ipv4`` is an object with two attributes, ``nat_1_1`` and ``vpc``; ``vpc`` is the VPC Subnet IPv4
+        address for this Interface, and ``nat_1_1`` is the 1:1 NAT IPv4 address, used to associate a public
+        IPv4 address with the VPC Subnet IPv4 address assigned to this Interface.
+
+        You can checkout the `interfaces schema on Linode API documentation`_ for details of behaviors of
```
Comment (lgarber-akamai) [inline doc/topics/cloud/linode.rst:128]: nitpick 🙂 
```suggestion
        You can check out the `interfaces schema on Linode API documentation`_ for details on the behaviors of
```

**Review by lgarber-akamai — APPROVED** [review]
LGTM! Tested with the following:

#### Integration testing:

```bash
python -m nox -e "test-3(coverage=False)" -- tests/integration/cloud/clouds/test_linode.py --run-expensive
```

#### Manual Testing:

Profile:

```yaml
linode-interfaces:
  provider: linode-provider
  size: g6-standard-1
  image: linode/ubuntu22.04
  location: us-mia
  password: T3stp4ssw0rd!!!1231235435
  assign_private_ip: true
  ssh_interface: private_ips
  interfaces:
    - purpose: public
      primary: true
    - purpose: vlan
      label: salt-test-vlan
      ipam_address: 172.0.0.3/32
    - purpose: vpc
      subnet_id: 20222  # VPC & subnet were created beforehand
      ipv4:
        vpc: 10.0.0.2
```

Creating an instance:

```
salt-cloud -p linode-interfaces salt-test-interfaces
```

File: doc/topics/cloud/linode.rst (hunk 1 above)
Comment (zliang-akamai) [reply doc/topics/cloud/linode.rst]: Thank you for the suggestion!

**Comment by zliang-akamai** [comment]
Hi @dwoz, may I get your review on this PR? Thank you so much! Let me know if there is anything needed.

**Comment by zliang-akamai** [comment]
Hi @dwoz, may I know if there is an estimated timeline for review and merge this PR? Thank you!

**Comment by zliang-akamai** [comment]
Hi @dwoz, I saw most third party cloud extensions were removed from master: https://github.com/saltstack/salt/pull/65971

I wonder if third party is no longer supported in 3008, may I rebase and get this PR merged into 3007 branch? And is there any plan to support these clouds in the path forward?

**Comment by displague** [comment]
@zliang-akamai I was wondering about the removal of extensions and the nearest I can tell the direction would be the one that the VMWare extension took:
- https://github.com/saltstack/salt-extension
- https://github.com/saltstack/salt-ext-modules-vmware

(I have used the `linode` cloud provider, possibly contributed, and am curious about the `packet` provider)

**Comment by dwoz** [comment]
@zliang-akamai 
Features should be merged into the master branch.

**Comment by zliang-akamai** [comment]
@dwoz I think Salt cloud module (which is edited in this PR) has been removed from `master`. In that case, I think we can close this PR?

**Pull request closed without merging**